packages/tools/examples/dynamicallyAddAnnotations/index.ts

```diff
@@ -0,0 +1,406 @@
+      ),
+      parseFloat(
+        (document.getElementById('start-y') as HTMLInputElement).value
+      ),
+      parseFloat(
+        (document.getElementById('start-z') as HTMLInputElement).value
+      ),
+    ];
+    const end = [
+      parseFloat((document.getElementById('end-x') as HTMLInputElement).value),
+      parseFloat((document.getElementById('end-y') as HTMLInputElement).value),
+      parseFloat((document.getElementById('end-z') as HTMLInputElement).value),
+    ];
+
+    addProgrammaticAnnotation(start, end, 'CT_STACK_AXIAL', null);
+  });
+
+document
+  .getElementById('add-world-coords-volume')
+  .addEventListener('click', () => {
+    const start = [
+      parseFloat(
+        (document.getElementById('start-x') as HTMLInputElement).value
+      ),
+      parseFloat(
+        (document.getElementById('start-y') as HTMLInputElement).value
+      ),
+      parseFloat(
+        (document.getElementById('start-z') as HTMLInputElement).value
+      ),
+    ];
+    const end = [
+      parseFloat((document.getElementById('end-x') as HTMLInputElement).value),
+      parseFloat((document.getElementById('end-y') as HTMLInputElement).value),
+      parseFloat((document.getElementById('end-z') as HTMLInputElement).value),
+    ];
+
+    addProgrammaticAnnotation(start, end, 'CT_VOLUME_AXIAL');
+  });
+
+const addProgrammaticAnnotation = (
+  start: number[],
+  end: number[],
+  viewportId: string,
+  type?: string
+) => {
+  const renderingEngine = getRenderingEngine(renderingEngineId);
+  const viewport = <Types.IStackViewport>(
+    renderingEngine.getViewport(viewportId)
+  );
+  if (type === 'image') {
+    // convert image coords to world coords
+    start = viewport.canvasToWorld(<Types.Point2>[...start]);
+    end = viewport.canvasToWorld(<Types.Point2>[...end]);
+  }
+  const referencedImageId = viewport.getCurrentImageId();
+  const FrameOfReferenceUID = viewport.getFrameOfReferenceUID();
+  const {
+    viewUp,
+    position: cameraPosition,
+    viewPlaneNormal,
+    focalPoint: cameraFocalPoint,
+  } = viewport.getCamera();
+
+  const annotation = {
+    highlighted: true,
+    invalidated: false,
+    metadata: {
+      toolName: 'Length',
+      referencedImageId,
+      viewUp,
+      cameraPosition,
+      FrameOfReferenceUID,
+      viewPlaneNormal,
+      cameraFocalPoint,
+    },
+    data: {
+      handles: {
+        points: [<Types.Point3>[...start], <Types.Point3>[...end]],
+        activeHandleIndex: null,
+        textBox: {
+          hasMoved: false,
+          worldPosition: <Types.Point3>[0, 0, 0],
+          worldBoundingBox: {
+            topLeft: <Types.Point3>[0, 0, 0],
+            topRight: <Types.Point3>[0, 0, 0],
+            bottomLeft: <Types.Point3>[0, 0, 0],
+            bottomRight: <Types.Point3>[0, 0, 0],
+          },
+        },
+      },
+      label: '',
+      cachedStats: {},
+    },
+    isLocked: true,
+    isVisible: true,
+  };
```

Comment:
Can we remove the unnecessary properties? For instance, do we need `label`? Do we need `cachedStats: {}`? Do we need `isLocked`? Do we need `isVisible`?

Reply:
done

this is the bare min

```ts
  const annotation = {
    metadata: {
      toolName: 'Length',
      referencedImageId,
      viewUp,
      cameraPosition,
      FrameOfReferenceUID,
      viewPlaneNormal,
      cameraFocalPoint,
    },
    data: {
      handles: {
        points: [<Types.Point3>[...start], <Types.Point3>[...end]],
        textBox: {},
      },
      cachedStats: {},
    },
  };
  ```

Reply:
textBox and cachedStats should be defined otherwise the tools tries to access undefined objects, rest is fine